#35: create webapisync option in cert utils

mohit-s96 opened the pull request:
closes #35 

### Example CLI command:
```bash
node index.js syncWebApiResults -p <path to results> -u http://localhost -r M00000533,M00000797 -i 50007 -o
```

Also fixes the lookup metadata issue.
cc: @darnjo

mohit-s96 committed:
#35: fixed error when fetching org data from new endpoint

mohit-s96 committed:
#35: fixed fixed lookup report error

also renamed files based on their use i.e. dd restore file is restore-utils/data-dictionary and web api sync util will be restore-util/web-api-core

mohit-s96 committed:
#35: added commander options and handler for the webApiSyncResults option

mohit-s96 committed:
#35: added syncWebAPiResults command and implementation

mohit-s96 committed:
#35: updated JSDoc types

mohit-s96 committed:
#35: fixed some typos and errors

mohit-s96 committed:
#35: changed input format

mohit-s96 committed:
Merge branch 'main' into 35-create-webapisync-option-in-cert-utils

mohit-s96 committed:
Merge branch 'main' into 35-create-webapisync-option-in-cert-utils

darnjo commented:
@mohit-s96 This needs to be updated with the changes from #34 - which no longer deletes reports but overwrites/archives them instead.

darnjo committed:
Merge branch 'main' into 35-create-webapisync-option-in-cert-utils

darnjo commented on data-access/cert-api-client.js:
```diff
@@ -46,6 +46,66 @@ const postDataDictionaryResultsToApi = async ({
+  }
+};
+
+const deleteDataDictionaryResults = async ({ url, reportId } = {}) => {
```
Deleting reports to add new ones was removed in #34 and there is no more `--overwrite` option.

We can leave this code here, but should hook it up to a `--deleteExistingReport` option instead.

mohit-s96 committed:
#35: removed overwrite and added deleteExistingReport option

mohit-s96 committed:
#35: changed process.exit to return

Since this package would be used in AWS/Cert API, we don't want to exit the node process. Thus we could just return in case of an error.

mohit-s96 committed:
Merge branch 'main' into 35-create-webapisync-option-in-cert-utils

mohit-s96 committed:
#35: restructure project like #43

darnjo commented on lib/misc/data-access/cert-api-client.js:
```diff
@@ -46,6 +46,66 @@ const postDataDictionaryResultsToApi = async ({
+  try {
+    const { id: reportId = null } =
+      (
+        await axios.post(
```
TODO: We should be using the Fetch API for everything at this point.

darnjo replied:
Can write the Auth part as a one-liner 

```
const getBearerTokenAuthHeader = (token = '') => (token?.length ? { Authorization: `Bearer ${token}` } : {});
```

Then call Fetch API like so:

```
const response = await fetch(requestUri, { headers });
```

darnjo commented:
Waiting on @darnjo to make a new PR with directories for data-dictionary and web-api-core under the restore-utils, with their own index.js and README.md files, and one at the root of the restore-utils which exports relevant items as libs.
